bundle/deploy/terraform/load.go

```diff
@@ -53,16 +52,13 @@ func (l *load) Apply(ctx context.Context, b *bundle.Bundle) diag.Diagnostics {
 	return nil
 }
 
-func (l *load) validateState(state *tfjson.State) error {
-	if state.Values == nil {
-		if slices.Contains(l.modes, ErrorOnEmptyState) {
-			return fmt.Errorf("no deployment state. Did you forget to run 'databricks bundle deploy'?")
-		}
-		return nil
+func (l *load) validateState(state *resourcesState) error {
+	if state.Version != nil && *state.Version != 4 {
```

Comment:
Why version 4? Can we define a named constant for it?

Reply:
That's the latest version terraform uses in it's state files - and the one we support right now. If if we update the terraform to some future version that produces state with a different version, we better know that and update the code to reflect the changes (if only just bumping the version)